account_sdk/src/webauthn_signer/account.rs

```diff
@@ -0,0 +1,176 @@
+use async_trait::async_trait;
+use serde::Serialize;
+use starknet::{
+    accounts::{
+        Account, Call, ConnectedAccount, Declaration, Execution, ExecutionEncoder,
+        LegacyDeclaration, RawDeclaration, RawExecution, RawLegacyDeclaration,
+    },
+    core::{
+        crypto::EcdsaSignError,
+        types::{
+            contract::legacy::LegacyContractClass, BlockId, BlockTag, FieldElement,
+            FlattenedSierraClass,
+        },
+    },
+    providers::Provider,
+};
+use std::sync::Arc;
+
+use crate::felt_ser::to_felts;
+
+use super::{cairo_args::VerifyWebauthnSignerArgs, P256r1Signer};
+
+pub struct WebauthnAccount<P>
+where
+    P: Provider + Send,
+{
+    provider: P,
+    // Later the struct will be generic over the signer type
+    // and will support "external" signers
+    signer: P256r1Signer,
+    address: FieldElement,
+    chain_id: FieldElement,
+    block_id: BlockId,
+    origin: String,
+}
+impl<P> WebauthnAccount<P>
+where
+    P: Provider + Send,
+{
+    pub fn new(
+        provider: P,
+        signer: P256r1Signer,
+        address: FieldElement,
+        chain_id: FieldElement,
+    ) -> Self {
+        Self {
+            provider,
+            signer,
+            address,
+            chain_id,
+            block_id: BlockId::Tag(BlockTag::Latest),
+            // Not security critical, but should be agreed upon
+            origin: "starknet".to_string(),
+        }
+    }
+}
+
+//starknet::accounts::Call really should implement serde::Serialize, at least as a crate feature
```

Comment:
Good call, can be a good issue for starknet-rs as surely more types may require this trait to be derived.